src/main/java/net/mcreator/element/types/MusicDisc.java

```diff
@@ -25,19 +25,21 @@
 import net.mcreator.element.parts.procedure.Procedure;
 import net.mcreator.element.parts.procedure.StringListProcedure;
 import net.mcreator.element.types.interfaces.IItem;
+import net.mcreator.element.types.interfaces.IItemWithTexture;
 import net.mcreator.element.types.interfaces.ITabContainedElement;
 import net.mcreator.minecraft.MCItem;
 import net.mcreator.ui.workspace.resources.TextureType;
 import net.mcreator.util.image.ImageUtils;
 import net.mcreator.workspace.elements.ModElement;
+import net.mcreator.workspace.references.TextureReference;
 
 import java.awt.image.BufferedImage;
 import java.util.List;
 
-public class MusicDisc extends GeneratableElement implements IItem, ITabContainedElement {
+public class MusicDisc extends GeneratableElement implements IItem, IItemWithTexture, ITabContainedElement {
 
 	public String name;
-	public String texture;
+	@TextureReference(TextureType.ITEM) public String texture;
 	public String description;
 	public TabEntry creativeTab;
 	public StringListProcedure specialInformation;

```

Comment:
Did not have time to test this part, so I will just ask instead, I guess those special Procedure types also get matched?

Reply:
RF checks if a value is same class or its subclass, so they should be after I fix your other notes.